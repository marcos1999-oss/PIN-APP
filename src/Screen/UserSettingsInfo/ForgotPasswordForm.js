import React from "react";
import { Text, View, SafeAreaView, Dimensions } from "react-native";
import { Card, CardItem, Body } from "native-base";
import { Formik } from "formik";
import CustomIcon from "../../CustomIcon";
import { Btn, TextInput } from "../../Components/Common";
import { scale, moderateScale, verticalScale } from "../../Utils/scaling";
import Image from "react-native-auto-height-image";
import { city2 } from "../../Assets";
import { FormContainer } from "./Common";
import * as authActions from "../../redux/actions/authActions";
import { connect } from "react-redux";
import { bindActionCreators } from 'redux';
import * as yup from "yup";
import { successMessage, errorMessage } from '../../Utils/alerts'
import { setCredentials } from '../../redux/services/api'

import style from "./style.js";

const validationSchema = yup.object().shape({
  password: yup
    .string()
    .required()
    .min(6)
    .label("Password"),
  cpassword: yup
    .string()
    .required()
    .min(6)
    .oneOf([yup.ref("password")], "Passwords must match", function (value) {
      return this.parent.password === value;
    })
    .label("Repeat Password"),
});


class ForgotPasswordForm extends React.Component {
  form = {
    fields: {
      password: '',
      cpassword: '',
    },
    refs: {},
  };

  state = {
    isLoading: false,
  };

  onSubmit = (values, _actions) => {
    this.setState({ isLoading: true });

    // TODO, everywhere this param is set (ie: isUserLoginForm), replace with a proper store/redux solution
    const isUserLoginForm = this.props.navigation.getParam(
      "isUserLoginForm",
      false
    );

    const userType = isUserLoginForm ? 'user' : 'business';
    const token = this.props.navigation.getParam('token');
    const password = values['password'];

    this.props.authActions.resetPasswordSave({
      userType,
      token,
      password,
      onSuccess: (access_token, client, uid) => {
        this.setState({ isLoading: false });

        const { navigation } = this.props;

        setCredentials(access_token, client, uid);

        this.props.authActions.fetchMe({
          userType,
          onSuccess: (response) => {
            // TODO, review. this call should be on Saga?
            this.props.authActions.signInSuccess({ response, access_token, client, uid });

            if (userType === 'user') {
              navigation.navigate('User');
            } else {
              navigation.navigate('Business');
            }
          },
          onFail: (error) => {
            errorMessage({ message: 'Error fetching user information', description: error.message });

            this.props.authActions.signOut({
              onSuccess: () => {
                navigation.navigate('Login');
              },
              onFail: (error) => {
                errorMessage({ message: 'Sign out failed', description: error.message });
              }
            });
          }
        });
      }, onFail: (error) => {
        this.setState({ isLoading: false });
        errorMessage({ message: 'Error when changing the password', description: error.errorMessage });
      }
    });
  };

  render() {
    const deviceWidth = Dimensions.get("window").width;
    const { isLoading } = this.state;

    return (
      <SafeAreaView style={style.l_container}>
        <View style={{ marginTop: 5, alignItems: "center", marginBottom: 15 }}>
          <CustomIcon name="setting" size={40} color={"#fff"} />
        </View>

        <React.Fragment>
          <Card style={{ borderRadius: verticalScale(7), }}>
            <CardItem style={{ borderRadius: verticalScale(7), }}>
              <Body>
                <Formik onSubmit={this.onSubmit} validationSchema={validationSchema} initialValues={this.form.fields}>
                  {(props) => (
                    <React.Fragment>
                      <TextInput
                        style={style.fontSize14}
                        secureTextEntry
                        form={props}
                        formKey="password"
                        value={props.values.password}
                        placeholder="Password"
                        setRef={(ref) => (this.form.refs.password = ref)}
                        onSubmitEditing={() =>
                          this.form.refs.cpassword._root.focus()
                        }
                      />
                      <TextInput
                        style={style.fontSize14}
                        secureTextEntry
                        form={props}
                        formKey="cpassword"
                        value={props.values.cpassword}
                        placeholder="Repeat Password"
                        setRef={(ref) => (this.form.refs.cpassword = ref)}
                        onSubmitEditing={props.handleSubmit}
                      />
                      <Text style={style.error}>
                        {props.errors && props.errors[Object.keys(props.errors)[0]]}
                      </Text>
                      <Btn
                        block
                        rounded
                        title="Save"
                        disabled={isLoading}
                        isLoading={isLoading}
                        onPress={props.handleSubmit}
                        style={{ color: '#25B7D3' }}
                        textColor='white'
                        textStyle={{ fontSize: moderateScale(14), fontWeight: "bold" }}
                      />
                    </React.Fragment>
                  )}
                </Formik>
              </Body>
            </CardItem>
          </Card>
        </React.Fragment>
      </SafeAreaView>
    );
  }
}

const mapStateToProps = (state) => {
  return {}
};

const mapDispatchToProps = (dispatch) => {
  return {
    authActions: bindActionCreators(authActions, dispatch),
  }
};

export default connect(mapStateToProps, mapDispatchToProps)(ForgotPasswordForm);
