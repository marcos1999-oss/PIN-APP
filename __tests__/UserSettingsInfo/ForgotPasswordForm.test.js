/**
 * @format
 * @lint-ignore-every XPLATJSCOPYRIGHT1
 */

import 'react-native';
import React from 'react';
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

import ForgotPasswordForm from '../../src/Screen/UserSettingsInfo/ForgotPasswordForm';

const mockStore = configureMockStore();

jest.mock('../../src/redux/services/config', () => {
  return {
    BASE_API_URL: 'http://127.0.0.1/v1'
  }
});

jest.mock('react-native-cached-images', () => {
  return {
    DocumentDir: () => {},
    ImageCache: {
      get: {
        clear: () => {},
      },
    },
  }
});

test('renders correctly', () => {
  const store = mockStore({
    auth: {
      me: {}
    },
    posts: {
      data: {},
    },
    offers: {
      data: {},
    },
    firms: {
      data: {},
    },
    features: {
      data: {},
    },
    holidays: {
      data: {},
    },
    schedules: {
      data: {},
    },
    pins: {
      near: {},
    }
  });
  const tree = renderer.create(
    <Provider store={store}>
      <ForgotPasswordForm />
    </Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
