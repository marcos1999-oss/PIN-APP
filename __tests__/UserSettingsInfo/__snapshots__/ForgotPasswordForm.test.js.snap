// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#5F92F3",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "marginBottom": 15,
        "marginTop": 5,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#fff",
            "fontSize": 40,
          },
          undefined,
          Object {
            "fontFamily": "PinAppIcon",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      î¤’
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
          "borderBottomWidth": 0.5,
          "borderColor": "#ccc",
          "borderLeftWidth": 0.5,
          "borderRadius": 2,
          "borderRightWidth": 0.5,
          "borderTopWidth": 0.5,
          "elevation": 3,
          "flexWrap": "nowrap",
          "marginBottom": 5,
          "marginLeft": 2,
          "marginRight": 2,
          "marginTop": 5,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.1,
          "shadowRadius": 1.5,
        },
        Object {
          "borderRadius": 14,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderRadius": 2,
            "flexDirection": "row",
            "paddingBottom": 10,
            "paddingLeft": 15,
            "paddingRight": 15,
            "paddingTop": 10,
          },
          Object {
            "borderRadius": 14,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "alignSelf": "stretch",
            "flex": 1,
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderBottomWidth": 1,
              "borderColor": "#D9D5DC",
              "borderLeftWidth": 0,
              "borderRadius": undefined,
              "borderRightWidth": 0,
              "borderTopWidth": 0,
              "borderWidth": undefined,
              "flexDirection": "row",
              "marginLeft": 2,
              "opacity": 1,
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            autoCapitalize="none"
            autoCorrect={false}
            editable={true}
            onChangeText={[Function]}
            onSubmitEditing={[Function]}
            placeholder="Password"
            placeholderTextColor="#000000"
            rejectResponderTermination={true}
            returnKeyType="next"
            secureTextEntry={true}
            style={
              Object {
                "color": "#000",
                "flex": 1,
                "fontSize": 17,
                "height": 50,
                "paddingLeft": 5,
                "paddingRight": 5,
                "top": 1.5,
              }
            }
            underlineColorAndroid="rgba(0,0,0,0)"
            value=""
          />
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderBottomWidth": 1,
              "borderColor": "#D9D5DC",
              "borderLeftWidth": 0,
              "borderRadius": undefined,
              "borderRightWidth": 0,
              "borderTopWidth": 0,
              "borderWidth": undefined,
              "flexDirection": "row",
              "marginLeft": 2,
              "opacity": 1,
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            autoCapitalize="none"
            autoCorrect={false}
            editable={true}
            onChangeText={[Function]}
            onSubmitEditing={[Function]}
            placeholder="Repeat Password"
            placeholderTextColor="#000000"
            rejectResponderTermination={true}
            returnKeyType="next"
            secureTextEntry={true}
            style={
              Object {
                "color": "#000",
                "flex": 1,
                "fontSize": 17,
                "height": 50,
                "paddingLeft": 5,
                "paddingRight": 5,
                "top": 1.5,
              }
            }
            underlineColorAndroid="rgba(0,0,0,0)"
            value=""
          />
        </View>
        <Text
          style={
            Object {
              "color": "#FF0000",
              "fontSize": 12,
              "paddingHorizontal": 10,
              "paddingVertical": 10,
            }
          }
        />
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "stretch",
              "backgroundColor": "#007aff",
              "borderBottomWidth": null,
              "borderColor": "#007aff",
              "borderLeftWidth": null,
              "borderRadius": 57,
              "borderRightWidth": null,
              "borderTopWidth": null,
              "borderWidth": undefined,
              "color": "#25B7D3",
              "elevation": 2,
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "center",
              "opacity": 1,
              "paddingBottom": 6,
              "paddingTop": 6,
              "shadowColor": undefined,
              "shadowOffset": undefined,
              "shadowOpacity": undefined,
              "shadowRadius": undefined,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 24,
                },
                Object {
                  "fontSize": 21,
                  "fontWeight": "bold",
                },
                Object {
                  "color": "white",
                },
              ]
            }
          >
            Save
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
